### PR TITLE
`rust_rules_workspace_name` now defaults to `rules_rust`

### PR DESCRIPTION
**Pull request by UebelAndre**
This is blocked by https://github.com/bazelbuild/rules_rust/pull/500 but once that's through, users using modern `rules_rust` will have their build files generated with the correct name.

### CODE REVIEW

**Comment by acmcarther** [comment]
I feel like we should be waiting some time before making this change but I'm not sure how long.

Do you have any thoughts (beyond "do it now", which I think is probably the wrong choice). Absent any better suggestion, I might go with the totally arbitrary option of "one month".

**Comment by UebelAndre** [comment]
This is all contingent upon https://github.com/bazelbuild/rules_rust/pull/500 going through which is a change that was concluded was worth doing but I don't seem to have a reviewer. Whatever the transition period is, I'd like to know by what date the switch will be made. I think I'd opened the original PR as a draft almost a month ago and the issue longer than that. My general frustration is that the maintainers of `rules_rust` are not active enough to properly communicate this sort of transition. However, given that there's also no release model and users are expected to work against master which has been extremely volatile. I think https://github.com/bazelbuild/rules_rust/issues/499 should serve as enough heads up for people to transition. The PR has the `sed` command I used to do the conversion which anyone can use. So I don't think this is a high risk change or would have a meaningful negative impact on users. So long as there's enough of a paper trail to get users to find the correct setting to use newer versions of `cargo-raze` with older versions of `rules_rust` then I think the change is good to happen without waiting a long time.

**Comment by acmcarther** [comment]
Rules rust is basically in the same state as this repo, except for two thing. First, it benefits (unlike this repo) from a few internal users who are driving development forward, contingent on the time commitment they can allocate, and second, it does *not* benefit (unlike this repo) from having the original developer still around to feel obligated to respond to PRs in a timely fashion. 

This might change soon, according to some stuff I've seen floating around, but I'm not certain.

---

I just realized that I LGTMd changing the default value to "rules_rust" here before you submitted the change to rules_rust. That seems broken.

The "io_bazel" naming scheme I think is historical, and we should probably stick with it because the rules_go folks have stuck with it. Also, changing it induces a ton of churn across the universe for low or zero gain.

**Comment by UebelAndre** [comment]
The nature of `rules_rust` and `cargo-raze` require that changes be made to one place first so that a change can safely be made in another. In this case, the ability to control the name of the repo was merged and with the confirmation from @mfarrugi that the rename was a reasonable thing to do, I opened a draft to communicate intent in `rules_rust` and then opened a PR in `cargo-raze` to support arbitrary names. Again, given the directive in https://docs.bazel.build/versions/master/skylark/deploying.html#workspace 

> If your rules belong to the bazelbuild organization, you must use rules_<lang> (e.g. rules_mockascript).

I understand that `rules_go` is a much larger and more used repository which would incur a lot of churn, the rust rules are small enough to support this kind of change.

**Comment by acmcarther** [comment]
Do you have some evidence that other existing rules have updated from "io_bazel_*" to "*"? What motivated making this change in the first place?

**Comment by UebelAndre** [comment]
> Do you have some evidence that other existing rules have updated from "io_bazel_*" to "*"? What motivated making this change in the first place?

https://github.com/bazelbuild/rules_python/blame/aa27a3fe7e1a6c73028effe1c78e87d2e7fab641/README.md#L19-L23

Again, I can only act on what's being communicated in the documentation/communications. I feel this is true in regards to statements made about separating things like the native python, java, and cc rules from core Bazel into the respective rules repositories. I like the idea that each repository in the `bazelbuild` namespace shares the same naming convention, but if the Bazel team decides this is not actually something that should be done, then again, I would say it should be reflected in documentation/communications and I'm happy to go about my day. I do have a vested interest in this though because I started this endeavor a month ago and want always want to do my best to finish anything I start. I think the risk of the rename is quite low. It's an easy fix and it's even easier for a maintainer to create a release just before the rename and communicate that it's happening (which I think is the right thing to do). I feel the change is justifiably low risk because I've also seen changes that change the minimum supported version of `rules_rust` jump from `0.27` to `3.0.0` in I think the span of a month (I'll have to go check the commit history to confirm). 

Just to have said it, I really care about both `rules_rust` and `cargo-raze` and want both projects to gain a bunch of users and contributors. But from what I've gathered over months of using and contributing to the rules, they're not widely used and they're not notably stable when it comes to APIs or compatibility. If this is true, then I was hoping to use that fact to aggressively make beneficial changes to each project in hopes that they might reach a point where they're more attractive to larger companies (understanding that at that time there can no longer be aggressive changes).

**Comment by mfarrugi** [comment]
`rules_rust` is better and more consistent, the real question here is how to deal with breaking changes. 

One option here is to deal w/ the backwards compatibility shims (assuming binds work for this), the other is to tag releases, and the last is to focus on head. I think a lot of the reason this project has done the latter is because upstream bazel moves very quickly, and most integration work is mired in breaking changes already.

I don't think this needs to be answered for this PR, and waiting is unlikely to benefit downstream.

**Comment by acmcarther** [comment]
Seeing rules_python make the change is convincing enough. Will approve and submit after the rules_rust change.

**Comment by acmcarther** [comment]
This LGTM but it needs a sync i think

**Comment by UebelAndre** [comment]
This is still blocked by the upstream repo not having been renamed. This is waiting on the release of Bazel 4.0.

**Comment by UebelAndre** [comment]
@acmcarther This is ready to go now that the upstream rules have been renamed

**Comment by UebelAndre** [comment]
@dfreese Also if you're around, could I get you to merge this as well if you're still around? 🙏

**Comment by UebelAndre** [comment]
@acmcarther Also, now that the alias issue has been merged, I think it'd be a good idea to enable BuildKite for PRs. This way I can work on this as well. I think I've solved for the non hermetic tests but it'll require a change from `rules_rust` which has been renamed and will require something like the repo rename done in this PR.

**Comment by UebelAndre** [comment]
FYI, the activity here has been me updating the version of `rules_rust`

**Comment by UebelAndre** [comment]
@acmcarther Sorry to keep pinging this issue, but it'd be a huge help if this could be merged and a new release cut. This is starting to cause some other PRs to get backed up

**Comment by UebelAndre** [comment]
@acmcarther Hey, really hoping to catch you before the weekend. If you can merge this and enable BuildKite for PRs, I can spend some time trying to fix it. For the time being I think it'd be fine to have it enabled and breaking now that GithubActions is in place. Maintainers can use best judgement looking at both builds

**Comment by UebelAndre** [comment]
@dfreese Hey! If you're around, can you merge this? I've nearly got CI working for BuildKite but I need modern `rules_rust` which is brought in by this change. 🙏 😅

**Comment by UebelAndre** [comment]
@acmcarther I guess I keep missing everyone 😅 

It'd help a lot to have this merged. I opened https://github.com/bazelbuild/rules_rust/pull/578 so I could have somewhere to test BuildKite CI and found a lot of really good info. Made some fixes but they're behind this PR. If I can merge this then I can open others. The Windows is still the only thing that doesn't work but it's now much farther than it used to. With more free time I'll have it solved

**Comment by UebelAndre** [comment]
@acmcarther Hey, just pinging this again, would really like to get a few PRs merged before the weekend so I can keep working on CI 🙏

**Comment by UebelAndre** [comment]
@dfreese Rebase resolved! Can this one get merged too? 🙏

**Comment by UebelAndre** [comment]
@acmcarther @dfreese Hey, just pinging this again 😅

**Comment by dfreese** [comment]
@UebelAndre I'll take a look at this in my afternoon.  It's probably reasonable to push this out and bump to 0.10.

**Comment by UebelAndre** [comment]
That would be amazing! If I can get this PR and a few others through, I'll have gotten BazelCI working so pretty excited to get it all in 😄

**Review by dfreese — APPROVED** [review]

File: README.md
Comment (UebelAndre) [reply README.md]: Ah, good call. Fixed

**Comment by dfreese** [comment]
now in v0.10.0

**Pull request merged**
